chiselFrontend/src/main/scala/chisel3/core/Data.scala

```diff
@@ -314,14 +314,18 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc { // sc
     _binding = Some(target)
   }
 
+  // Similar to topBindingOpt except it explicitly excludes SampleElements which are bound but not
+  // hardware
+  private[core] final def isSynthesizable: Boolean = _binding.map {
```

Comment:
Currently we define properties of the different bindings in the Binding type hierarchy, can we introduce a SynthesizableBinding trait and check for that, instead of scattering binding definition logic throughout?

Reply:
I thought about such a refactoring but it seems like a large task for a bug fix, we need this bug fixed to bump rocket-chip, and the code is strictly more explicit than it was before